File: app/src/main/java/com/example/groupgetter/SubredditSearchResponse.java
package com.example.groupgetter;

import java.util.List;

public class SubredditSearchResponse {

    private List<SubredditSearchResponse.Subreddit> data;
    public List<SubredditSearchResponse.Subreddit> getData() {
        return data;
    }

    public class Subreddit{
        private String display_name;
        public String getDisplay_name() {
            return display_name;
        }
    }
}

File: app/src/main/java/com/example/groupgetter/SubredditSearchResponse.kt
package com.example.groupgetter

//data classes representing the structure of a subreddit search response
data class SubredditSearchResponse(
        val data: SubredditsInfo
)

data class SubredditsInfo(
        val children: List<SubredditChild>
)

data class SubredditChild(
        val data: SubredditInformation
)

data class SubredditInformation(
        val display_name: String,
        val title: String,
        val description: String,
        val icon: String,
)

